test/helpers/kubectl.go

```diff
@@ -1681,13 +1681,26 @@ func (kub *Kubectl) overwriteHelmOptions(options map[string]string) error {
 		if err != nil {
 			return err
 		}
+		devices := privateIface
 
 		opts := map[string]string{
 			"global.kubeProxyReplacement": "strict",
-			"global.nodePort.device":      privateIface,
 			"global.k8sServiceHost":       nodeIP,
 			"global.k8sServicePort":       "6443",
 		}
+
+		if RunsOnNetNextOr419Kernel() {
+			// Enable BPF masquerading
+			defaultIface, err := kub.GetDefaultIface()
+			if err != nil {
+				return err
```

Comment:
Should this log an error message? Asking because tests below are removed in another commit, so it might be more difficult to troubleshoot if for some reason we fail to get the default interface?

``` diff
-		defaultIface, err = kubectl.GetDefaultIface()
-		Expect(err).Should(BeNil(), "Cannot determine default iface")
```

Reply:
I think we should not put assertions into helper functions, and assert only in consumers of those helpers. E.g. in this case if the default iface cannot be determined, then assertions in `DeployCiliumAndDNS()` will fail with the msg saying something similar to "cannot deploy cilium because default iface cannot be determined".

Reply:
Ok, fair enough :+1: